### PR TITLE
Introduce SkipBlocksWithOutOfOrderChunksEnabled feature

### LINKED ISSUE #4692: Compactor - skipping out-of-order blocks feature is disabled
There is a PR to introduce Thanos feature of skipping out of order blocks https://github.com/cortexproject/cortex/pull/4453. It was raised late August 2021 and has stayed open ever since. 
However there was yet another PR raised later (I believe "later" being an important factor here), namely in October 2021 https://github.com/cortexproject/cortex/pull/4505 that got merged and introduced that feature, however hardcoded it to be disabled: https://github.com/cortexproject/cortex/blob/release-1.11/pkg/compactor/compactor.go#L660. 

Why is that so? Is the code only partially ready for this feature to be enabled? What would be required in order for that feature to be enabled for Cortex? I believe there are at least several issues / people that would benefit from it (myself included).

Thanks.

### PR DESCRIPTION
**Pull request by alvinlin123**
Signed-off-by: Alvin Lin <alvinlin@amazon.com>

**What this PR does**:
This PR allows user to configure if they want to skip out of order chunks during compaction (special thanks to @alanprot working on https://github.com/cortexproject/cortex/pull/4706 concurrently). 

This PR also allows manual creation of `no-compact-mark.json`. 

**Which issue(s) this PR fixes**:
Fixes #4692
Partially addresses #4705 - this change will allow manual creation of `no-compact-mark.json`

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,533 +2,534 @@
 
 ## master / unreleased
 
-* [FEATURE] Ruler: Add `external_labels` option to tag all alerts with a given set of labels.
```
Comment (alanprot) [inline CHANGELOG.md:5]: seems that those changes were not intentional

Comment (alvinlin123) [reply CHANGELOG.md:5]: Oh this one is intentional because FEATURE should go after CHANGE (I as added the PR number)

Comment (alvinlin123) [reply CHANGELOG.md:5]: I will fix the unintended change (from '*' to '-' four bullets)

**Review by alanprot — APPROVED** [review]

**Comment by alanprot** [comment]
LGTM

**Pull request merged**